### PR TITLE
Documentation for v4.0.0 including migration guide

### PR DESCRIPTION
**Pull request by boesing**
|    Q          |   A
|-------------- | ------
| Documentation | yes

### Description

This introduces initial `v4` documentation including an initial migration guide. This is work in progress and thus might change with other PRs in the future.

Adapter modifications can be reviewed here:
https://github.com/laminas/laminas-cache/pull/274/commits/51433fb3e04ceef4ff2dc79f98eb8ce8d4f03fdf

### CODE REVIEW

**Comment by boesing** [comment]
Will rebase this once #272 is merged.

**Comment by froschdesign** [comment]
I will review this and add some commits.

File: docs/book/v4/storage/adapter.md [hunk 1]
```diff
@@ -0,0 +1,1015 @@
...
+
+    /**
+     * Increment an item.
+     *
+     * @param  string $key
+     * @param  int    $value
+     * @return int|bool The new value on success, false on failure
+     * @throws \Laminas\Cache\Exception\ExceptionInterface
+     */
+    public function incrementItem($key, $value);
```
Comment (boesing) [inline docs/book/v4/storage/adapter.md]: Remove once #117 is being implemented

File: docs/book/v4/storage/adapter.md [hunk 2]
```diff
@@ -0,0 +1,1015 @@
...
+
+    /**
+     * Increment multiple items.
+     *
+     * @param  array $keyValuePairs
+     * @return array Associative array of keys and new values
+     * @throws \Laminas\Cache\Exception\ExceptionInterface
+     */
+    public function incrementItems(array $keyValuePairs);
```
Comment (boesing) [inline docs/book/v4/storage/adapter.md]: Remove once #117 is being implemented

File: docs/book/v4/storage/adapter.md [hunk 3]
```diff
@@ -0,0 +1,1015 @@
...
+
+    /**
+     * Decrement an item.
+     *
+     * @param  string $key
+     * @param  int    $value
+     * @return int|bool The new value on success, false on failure
+     * @throws \Laminas\Cache\Exception\ExceptionInterface
+     */
+    public function decrementItem($key, $value);
```
Comment (boesing) [inline docs/book/v4/storage/adapter.md]: Remove once #117 is being implemented

File: docs/book/v4/storage/adapter.md [hunk 4]
```diff
@@ -0,0 +1,1015 @@
...
+
+    /**
+     * Decrement multiple items.
+     *
+     * @param  array $keyValuePairs
+     * @return array Associative array of keys and new values
+     * @throws \Laminas\Cache\Exception\ExceptionInterface
+     */
+    public function decrementItems(array $keyValuePairs);
```
Comment (boesing) [inline docs/book/v4/storage/adapter.md]: Remove once #117 is being implemented

**Comment by boesing** [comment]
- [x] Adapt inline codesnippets with changes of #295

**Comment by boesing** [comment]
- [x] Add BC break regarding `PluginOptions#getSerializer` does not create a serializer anymore if a string was passed beforehand

**Pull request merged**